### PR TITLE
drivers: CAN: Limit the DLC to 8

### PR DESCRIPTION
**Pull request by alexanderwachter**
This commit limits the data length code to eight.
DLC > 8 returns a newly introduced CAN_TX_EINVAL error code.

### CODE REVIEW

**Comment by alexanderwachter** [comment]
ping @nixward

File: include/drivers/can.h [hunk 1]
```diff
@@ -47,6 +47,9 @@ extern "C" {
 /** unexpected error */
 #define CAN_TX_UNKNOWN  (-5)
 
+/** invalid parameter */
+#define CAN_TX_EINVAL   (-22)
+
```
Comment (henrikbrixandersen) [inline include/drivers/can.h]: Was this change intended to be included in this PR?

Comment (henrikbrixandersen) [inline include/drivers/can.h]: Why not just use -EINVAL?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: include/drivers/can.h (hunk 1 above)
Comment (alexanderwachter) [reply include/drivers/can.h]: Because we don't use POSIX error codes in CAN, and this way, we can guarantee that the EINVAL doesn't overlap with some other CAN error code by accident. 22 is actually EINVAL

**Review by nixward — APPROVED** [review]

**Review by karstenkoenig — APPROVED** [review]
Looks good, thanks for keeping the different drivers in sync :+1:

File: include/drivers/can.h (hunk 1 above)
Comment (henrikbrixandersen) [reply include/drivers/can.h]: OK.

**Review by henrikbrixandersen — APPROVED** [review]
